pages/components/cards.js

```diff
@@ -0,0 +1,201 @@
+import React, { useState, useEffect } from "react";
+import Image from "next/image";
+import styled from "styled-components";
+
+const dummyPosts = [
+  {
+    id: 1,
+    content: "Dies ist der Inhalt des ersten Beitrags.",
+    Photo: "/assets/images/1.jpg",
+    comments: [],
+  },
+  {
+    id: 2,
+    content: "Dies ist der Inhalt des zweiten Beitrags.",
+    Photo: "/assets/images/2.jpg",
+    comments: [],
+  },
+  {
+    id: 3,
+    content: "Dies ist der Inhalt des dritten Beitrags.",
+    Photo: "/assets/images/3.jpg",
+    comments: [],
+  },
+];
+
+export default function Cards() {
+  const [posts, setPosts] = useState(dummyPosts);
+  const [isCommentVisible, setIsCommentVisible] = useState(false);
+
+  // to update the comment
+  const handleCommentUpdate = (event, postId) => {
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return {
+          ...post,
+          comment: event.target.value,
+        };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+  };
+
+  // comment Icon visible
+  const handleCommentClick = (postId) => {
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return {
+          ...post,
+          isCommentVisible: !post.isCommentVisible,
+        };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+  };
+
+  // comment function
+  const handleCommentSubmit = (event, postId) => {
+    event.preventDefault();
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        const updatedComments = [...post.comments, post.comment];
+        return { ...post, comments: updatedComments, comment: "" };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+  };
+
+  // likestatus Icon toggle
+  const handleLikeClick = (postId) => {
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return {
+          ...post,
+          likeStatus: !post.likeStatus,
+        };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+  };
```

Comment:
This feature is normally handled in a separate user story. Please consider this when writing the next user stories.